### ISSUE TITLE
Can't Control without tuya cloud

### ISSUE THREAD

**Issue by meluvalli** [jasonacox/tinytuya#461]
Hello.  I have a device that will only get dps once it connects to the cloud.

If I run :
```
data = d.status() 
print('set_status() result %r' % data)
```
My output if I block my device from the internet is:
```
set_status() result {'dps': {}, 'type': 'query', 't': 179}
```

But if I unblock it then it changes to:
```
set_status() result {'dps': {'12': False, '14': 0, '15': 0, '19': True, '101': 'ac_power_on', '103': 'real_time_mode', '112': 3}}
```

I blocked it again, rebooted it, then ran monitor on it and this is what it shows:
```
Received Payload: None
 > Send Heartbeat Ping <
Received Payload: {'dps': {'1': ''}, 't': 884}
 > Send Heartbeat Ping <
Received Payload: {'dps': {'3': 1}, 't': 884}
 > Send Heartbeat Ping <
Received Payload: {'dps': {'19': True}, 't': 884}
 > Send Heartbeat Ping <
Received Payload: {'dps': {'103': 'real_time_mode'}, 't': 884}
 > Send Heartbeat Ping <
Received Payload: {'dps': {'112': 3}, 't': 884}
 > Send Heartbeat Ping <
Received Payload: {'dps': {'12': False}, 't': 884}
 > Send Heartbeat Ping <
Received Payload: {'dps': {'14': 0}, 't': 884}
 > Send Heartbeat Ping <
Received Payload: {'dps': {'101': 'ac_power_on'}, 't': 884}
 > Send Heartbeat Ping <
Received Payload: None
 > Send Heartbeat Ping <
Received Payload: {'dps': {'11': 0}, 't': 884}
 > Send Heartbeat Ping <
 ```
 
 Any idea how to get dps's without connecting it to Tuya cloud?

**Comment by jasonacox** [comment]
Hi @meluvalli - Can you post what type of device this is (and link to a details about it if you can). That will help others who have  a similar device.

Keep in mind that Tuya devices are designed to connect to the cloud. That is their primary function. The local access point that many of the Tuya devices have, and TinyTuya uses for local status/control, is a secondary feature. You will need cloud access to register them and to get the local access key. Some devices, once registered, will work perfectly fine offline (no internet), others will refuse to work at all.

For your device, you are getting some updates during the monitor loop so at least it isn't fully disabled. You could try sending the updatedps command to see if that helps:

```python
d.updatedps(['18','19','20'], nowait=True)
```

**Comment by meluvalli** [comment]
Well, apparently after more research, some devices doesn't advertise the DPS correctly until the entity has been properly initialized onto Tuya server...  So, i'm guessing no way around this at this point.  I tried the d.updatedps, but didn't change anything.  Device still reports no dps's while offline.

This device in question is a pet feeder.  Below is the device information:
```
[
    {
        "name": "Pet Feeder",
        "id": "XXXXXXXXXXXXX",
        "key": "XXXXXXXXXXXXX",
        "mac": "XX:XX:XX:XX:XX:XX",
        "uuid": "XXXXXXXXXXXXX",
        "sn": "XXXXXXXXXXXXX",
        "category": "cwwsq",
        "product_name": "XTUOES",
        "product_id": "ibxe5jf6mtqmdfxl",
        "biz_type": 0,
        "model": "",
        "sub": false,
        "icon": "https://images.tuyaus.com/smart/icon/bay16227127058150Bvl/cce4b4106abc9a283c7add8cad7f6e5b.png",
        "mapping": {
            "1": {
                "code": "meal_plan",
                "type": "Raw",
                "values": {}
            },
            "3": {
                "code": "manual_feed",
                "type": "Integer",
                "values": {
                    "unit": "",
                    "min": 1,
                    "max": 20,
                    "scale": 0,
                    "step": 1
                }
            },
            "9": {
                "code": "factory_reset",
                "type": "Boolean",
                "values": {}
            },
            "11": {
                "code": "battery_percentage",
                "type": "Integer",
                "values": {
                    "unit": "%",
                    "min": 0,
                    "max": 100,
                    "scale": 0,
                    "step": 1
                }
            },
            "12": {
                "code": "charge_state",
                "type": "Boolean",
                "values": {}
            },
            "15": {
                "code": "feed_report",
                "type": "Integer",
                "values": {
                    "unit": "",
                    "min": 0,
                    "max": 20,
                    "scale": 0,
                    "step": 1
                }
            },
            "19": {
                "code": "light",
                "type": "Boolean",
                "values": {}
            }
        },
        "ip": "XX.XX.XX.XX",
        "version": "3.3"
    }
]
```

**Comment by meluvalli** [comment]
What's strange about it, is that I can still control the device using
```
data = d.set_status(3, 3)
```
It still does the feeder even though the dps isn't there.   And even after that, it the dps still isn't there!

**Comment by jasonacox** [comment]
Did you try sending updatedps() with that index? I suspect it won't make a difference, but worth trying.

```python
d.updatedps(['3'], nowait=True)
```

**Comment by meluvalli** [comment]
Yep.  No difference :(
```
set_status() result {'dps': {}, 'type': 'query', 't': 1474}
```

**Comment by jasonacox** [comment]
> set_status()

Did you mean `status()`? You're not going to get full DPS response with `set_status()` as that function is used to set DPS values rather than read them.

**Comment by meluvalli** [comment]
Yes.  Sorry, just coppied and pasted from another script :)
```
data = d.status() 
print('set_status() result %r' % data)
```
Was the code I used.  It is actually d.status :)

**Comment by lankhaar** [comment]
Hi there! It's been a while since this post had some activity, but where you able to fix the issue? 😅

**Comment by meluvalli** [comment]
Still doesn’t work for me.  Needs to connect to there server before it gets
it parameters :(

You can still control the device and send feed commands to it, you just
don’t get any confirmation or information from the device.

**Comment by lankhaar** [comment]
Yeah truee, same for me. It works, but since it won't update it's status, it'll always show "charging", even when it's cleaning.

Main issue is though that even the frickin battery level can't update without the cloud, so I'll never know when the battery is full/empty.

**Comment by jasonacox-sam** [comment]
**📋 Summary:** A Tuya pet feeder returns empty DPS (`{'dps': {}}`) when blocked from the internet, but populates all DPS once connected to Tuya Cloud. Commands like `set_status()` still work offline — but `status()` returns nothing useful without cloud connectivity. Multiple users confirmed the same behavior across different pet feeders.

This issue has been inactive for over 6 months, so we're closing it to keep things manageable. If you happened to find a solution, we'd love to hear about it — your finding could help others facing the same thing! Feel free to **reopen** this issue and share what worked, or open a new issue if you need further help.

Thank you for using TinyTuya! 🙏
